### ISSUE TITLE
pochhammer doesn't survive kill(all)

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#2322]
Imported from SourceForge on 2024-07-05 23:40:51
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2023-06-05 17:31:38
Original: https://sourceforge.net/p/maxima/bugs/4145

---

pochhammer doesn't work after kill(all), due to an autoloaded file not being reloaded.

this causes a test suite failure in rtest_orthopoly (in share).

~~~
(%i1) pochhammer(x,n);
(%o1)                                (x)
                                        n
(%i2) diff(pochhammer(x,n), x);
(%o2)                    (x)  (psi (x + n) - psi (x))
                            n     0             0
(%i3) :lisp *autoloaded-files*

(($POCHHAMMER . orthopoly))
(%i3) kill(all);
(%o0)                                done
(%i1) pochhammer(x,n);
(%o1)                          pochhammer(x, n)
(%i2) diff(pochhammer(x,n), x);
                             d
(%o2)                        -- (pochhammer(x, n))
                             dx
(%i3) :lisp *autoloaded-files*

NIL

~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 23:40:53
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2023-06-13 00:35:42
Original: https://sourceforge.net/p/maxima/bugs/4145/#07fd

---

I think `pochhammer` gets killed because it's on the `props` infolist, and it's on `props` because of a call to `$declare`to declare it `complex` (line 260 at present).

One workaround is to remove `pochhammer` from `props` immediately afterwards, although I think we should check to see if the declaration is actually needed -- if the declaration is removed, does it change any test results? I didn't try it. Lots of functions can possibly be complex, but I suspect many are not declared complex ...

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 23:40:56
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2023-06-13 19:43:45
Original: https://sourceforge.net/p/maxima/bugs/4145/#41ba

---

Thanks for the suggestion.

Fixed by [76f440c896ceba3df0fb9aecd9947a134c695a31]

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 23:41:00
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2023-06-13 19:55:41
Original: https://sourceforge.net/p/maxima/bugs/4145/#16fa

---

- **status**: open --> closed
